DataFormats/MuonReco/interface/MuonEnergy.h

```diff
@@ -49,13 +51,36 @@ namespace reco {
        /// DetId of the central HCAL tower with smallest depth
        DetId hcal_id;
        
+       ///
+       /// RecHits
+       ///
+       /// crossed HCAL rechits
+       const std::vector<HBHERecHit>* getHadRecHits() const
+       {
+	 if (hadRecHitsValid) return &hadRecHits;
+	 return nullptr;
+       }
+		 
+       void setHadRecHits(const std::vector<const HBHERecHit*>& rechits)
+       {
+	 hadRecHitsValid = true;
+	 hadRecHits.clear();
+	 for (auto hit: rechits)
+	   hadRecHits.push_back(*hit);
+       }
+
        MuonEnergy():
        tower(0), towerS9(0),
        em(0), emS9(0), emS25(0), emMax(0),
        had(0), hadS9(0), hadMax(0),
-	   ho(0), hoS9(0), ecal_time(0), ecal_timeError(0), hcal_time(0), hcal_timeError(0)
+	 ho(0), hoS9(0), ecal_time(0), ecal_timeError(0), hcal_time(0), hcal_timeError(0),
+	 hadRecHitsValid(false)
 	 { }
-       
+
+    private:
+       /// crossed HCAL rechits
+       std::vector<HBHERecHit> hadRecHits;

```

Comment:
It may help to add "Crossed" to the name so that there is possibly less confusion with what goes to hadS9 etc

Reply:
Long term we should know better what's needed. For now it's easier to have the whole rechit since they are not easily available. Also out of all data members we will use a number of them, i.e. time, id, energy, chi2, raw energy. 
Regarding the names, I'll add crossed everywhere to avoid confusion.

Reply:
>  For now it's easier to have the whole rechit

1% of miniAOD for just convenience is an overkill, considering that all the data is already available in the event.
In the past much smaller increments in event size required more optimal updates in the event content.
I see how a DetId may be not easy to provide simple lookup. Instead of that, a ValueMap with pointers to the interesting rechits should provide a simple way to access at a much smaller fraction of the cost.

@fgolf @peruzzim

Reply:
Not sure I understand the comment. Do we store HBHERecHits in MiniAOD?